user_tools/src/spark_rapids_tools/tools/qualx/qualx_main.py

```diff
@@ -442,22 +442,19 @@ def predict(
                 'app_meta': {},
                 'platform': platform,
             }
-            # search sub directories for appIds
-            appIds = find_paths(
-                metrics_dir, lambda x: RegexPattern.appId.match(x), return_directories=True
-            )
-            appIds = [Path(p).name for p in appIds]
-            if len(appIds) == 0:
+            # search sub directories for App IDs
+            app_ids = [p.name for p in Path(metrics_dir).iterdir() if p.is_dir()]
+            if len(app_ids) == 0:
```

Comment:
This fix should do it.
The trick part in this change that we have to rely on directory structure to pick the right folders.
For example, moving fwd, we can't introduce a subdirectory in raw-metrics  because it will be mistakenly treated as an Application folder.

Reply:
Thanks @amahussein. Yes, If we change the raw metrics folder then all its usages should be updated as well.